DOCS-11236-db-xa-datasource-kt (#1670)
mulesoft/docs-connectors: db/1.10/modules/ROOT/pages/database-connector-connection.adoc

</db:config>
----

> == Configure a Data Source Reference Connection when in XA Transaction Mode

Reviewer: ```suggestion
== Configure a Data Source Reference Connection When in XA Transaction Mode
```

> If your Mule app requires operations bound to XA transactions, and the connection pooling is enabled for the database configuration in use, the default connection pool implementation for the Data Source Reference Connection is Enhydra XAPool.

Reviewer: ```suggestion
If your Mule app requires operations bound to XA transactions, and connection pooling is enabled for the database configuration in use, the default connection pool implementation for Data Source Reference Connection is Enhydra XAPool.
```

> If your Mule app uses Database Connector operations such as *Insert*, *Select* or others that run with an XA transaction, and connection pooling is enabled via the *Pooling profile* fields of the connector configuration, the default connection pool implementation for the Data Source Reference Connection is Enhydra XAPool.

Reviewer 2: I'd say "If your Mule App requires operations bound to XA transactions, and the connection pooling is enabled for the database configuration in use .."

> Enhydra XAPool does not support prepared statements. To execute these operations use the Bitronix connection pool. Refer to xref:mule-runtime::using-bitronix-to-manage-transactions.adoc#configuring-the-bitronix-transaction-manager[configuring the Bitronix Transaction Manager] documentation for details.

Reviewer 3: The first word in the link should be capitalized - Configuring

> Note that Enhydra XAPool does not support prepared statements. To execute these operations use the Bitronix connection pool, refer to xref:mule-runtime::using-bitronix-to-manage-transactions.adoc#configuring-the-bitronix-transaction-manager[configuring the Bitronix Transaction Manager] documentation for details.

Reviewer: ```suggestion
Note that Enhydra XAPool does not support prepared statements. To execute these operations, use the Bitronix connection pool. Refer to xref:mule-runtime::using-bitronix-to-manage-transactions.adoc#configuring-the-bitronix-transaction-manager[configuring the Bitronix Transaction Manager] documentation for details.
```

> [NOTE]

Reviewer: Is it odd to follow a paragraph beginning with "Note that..." with an actual [NOTE]-formatted comment?
Just asking; not sure, myself, one way or the other.

When using XA transactions, you must provide an `XADataSource` object in the JDBC data source spring bean configuration file of your app.

> The following example shows how to configure a data source reference connection for a Derby database when in XA transaction mode:

Reviewer: Why is "a data source reference connection" lowercase here but not in Line 66? 
(Which is why I suggested removing "the" before the proper noun....)
Standardize.

> . In Studio, in the `src/main/resources` folder, create a `datasource-config.xml` bean file.

Reviewer: ```suggestion
. In Studio, in the `src/main/resources` folder, create a `datasource-config.xml` spring bean file.
```

> . Add the `XADataSource` object at the end of the class `org.enhydra.jdbc.standard.Standard`, for example `class="org.enhydra.jdbc.standard.StandardXaDataSource`.

Reviewer: ```suggestion
. Add the `XADataSource` object to the end of the class `org.enhydra.jdbc.standard.Standard`, for example, `class="org.enhydra.jdbc.standard.StandardXaDataSource`:
```

+
.datasource-config.xml File
[source,xml,linenums]
----
<?xml version="1.0" encoding="UTF-8"?>
<beans xmlns="http://www.springframework.org/schema/beans"
       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
             xsi:schemaLocation="
               http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-current.xsd">

    <bean id="jdbcDataSource" class="org.enhydra.jdbc.standard.StandardXaDataSource" destroy-method="shutdown">
        <property name="driverName" value="org.apache.derby.jdbc.EmbeddedDriver"/>
        <property name="url" value="jdbc:derby:muleEmbeddedDB;create=true"/>
    </bean>

</beans>
----
+

> . Return to your Mule flow, go to *Global Elements* > *Create* to open the *Choose Global Type* window.

Reviewer: ```suggestion
. Return to your Mule flow and navigate to *Global Elements* > *Create* to open the *Choose Global Type* window.
```

. In *Filter*, type `spring`, select *Spring Config*, and click *OK*.

> . In *Spring Config* > *Files*, add `datasource-config.xml` to reference the spring configuration file with the Spring module.

Reviewer: "spring" > "spring bean"?
You know best; please just search for and check that _spring, bean_, and _spring bean_ are used consistently in this topic set.

. Click *OK*.

image::database-spring-configuration.png[Spring Configuration window with datasource-config.xml configure in Files field]

[start=8]
. In *Global Elements*, click *Create* to open the *Choose Global Type* window.
. In *Filter*, type `database`, select *Database Config*, and click *OK*.
. In *Database Config* > *Connection*, select *Data Source Reference Connection*.
. Click *Configure* to add the required JDBC driver and select either: +
+
* *Use local file* +
Install the library using a local file.
* *Add Maven dependency* +
Install a Maven dependency to add to the project.
+

> . For *Data source ref*, add the bean id `jdbcDataSource` created in the `datasource-config.xml` bean file.

Reviewer: ```suggestion
. For *Data source ref*, add the bean ID `jdbcDataSource` created in the `datasource-config.xml` bean file.
```

> . In the *Advanced* tab, configure the *Pooling profile*, *Column types* and *Reconnection* fields.

Reviewer: ```suggestion
. In the *Advanced* tab, configure the *Pooling profile*, *Column types*, and *Reconnection* fields.
```
